source/slang/core.meta.slang

```diff
@@ -1210,6 +1210,31 @@ extension matrix<int16_t,R,C,L>
     __init(int value) { this = matrix<T,R,C,L>(T(value)); }
 }
 
+
+__generic<T>
+extension vector<T, 4>
+{
+    __generic<let L : int>
+    __implicit_conversion($(kConversionCost_MatrixToVector))
+    [__unsafeForceInlineEarly]
+    __init(matrix<T, 2, 2, L> value)
+    {
+        this.xy = value[0];
```

Comment:
Instead of doing this, which lowers to swizzledSet, just lower it to makeVector(elementExtract, elementExtract)